### PR TITLE
docs(todos demo): gives warning to new users of unimplemented functionality resulting in `POST /todos` failure

### PR DESCRIPTION
**Pull request by garethhcoleman**
As getFIlterJsonSchemaFor() is not yet Model-Aware, filter options are unset and default to
'String'. This one line note in the tutorial warns users and links in the issue for more information.

issue #4977

first PR - worked hard to follow guidance - hope it's correct.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by nabdelgadir** [comment]
Hi @garethhcoleman, thanks for the PR! I looked through the issue https://github.com/strongloop/loopback-next/issues/4977 and it seems like they're still trying to reproduce it, so I think we can put this PR on hold until it's reproduced.

Other notes just for future contributions:
1) The code lint is failing, which you can fix by running `npm run lint:fix`
2) Commit lint is also failing, which you can fix by following the guidelines [here](https://loopback.io/doc/en/lb4/code-contrib-lb4.html#commit-message-guidelines). In your case, it could be something like `fix(docs): add warning for filter errors in todo example`.

**Comment by garethhcoleman** [comment]
Thanks @nabdelgadir for considering this PR (my first against a 'real' project, hence noob errors).

Apologies for the linting fails, I realised the code lint tests were failing but as I wasn't changing any code and marked the fix in the commit as docs I thought it wouldn't matter. I also tried hard to get the commit message right including using commitizen but still failed. If you can spare a minute to help explain a couple of things I would be very grateful (and it may even help someone else).

Why do you suggest `fix` over `docs` for the **type** of commit?

And I also missed the requirement that **scope** 
> "must be a list of one or more packages contained in this monorepo"

Our of interest I can't see the package 'docs' [here](https://github.com/strongloop/loopback-next/tree/master/packages), can you explain why this is the correct choice for scope? Many thanks in advance!

**Comment by achrinza** [comment]
Hi @garethhcoleman, to answer your queries:

> Why do you suggest fix over docs for the type of commit?

AFAIK, it's due to the fact that the documentation is a Node.js package in itself (see below). Hence, in that context, it makes sense to use `fix` and `feat` as they're more descriptive than having all documentation changes be `docs`.

`docs` is used more for changes to [tsdoc](https://github.com/Microsoft/tsdoc).

> Our of interest I can't see the package 'docs' here, can you explain why this is the correct choice for scope?

In the LoopBack 4 monorepo, there's several globs where Node.js packages are stored (see [Lerna configuration](https://github.com/strongloop/loopback-next/blob/fb57e23a8aeab69b99928dafbc8cf9eb343cf36a/lerna.json#L3-L11)):

- acceptance/*
- benchmark
- docs
- examples/*
- packages/*
- extensions/*
- sandbox/*

One of those globs is the `docs` directory which is a [package in itself, and it's named `@loopback/docs`](https://github.com/strongloop/loopback-next/blob/master/docs/package.json#L2). It is also [published to NPM](https://www.npmjs.com/package/@loopback/docs). Hence that's where the `docs` scope comes from.

---

Feel free to voice out if there's anymore queries, we'll try our best to respond.

**Comment by garethhcoleman** [comment]
Thank you very much for the reply, there is so much to learn - I suppose a more experienced developer would have known that the package list referenced in the commit message guidance was incomplete because it missed the lerna config ... Likewise I suppose most programmers would know that the doc type is just for the tsdocs package and not for the docs package without needing to be reminded in the guidance!

**Comment by bajtos** [comment]
Thank you @garethhcoleman for the pull request! Don't worry about getting it right on the first time, we are happy to help you along your learning journey!

Instead of adding a warning to the documentation, perhaps we can fix the code creating filter schema to produce a valid default value? See https://github.com/strongloop/loopback-next/issues/4977#issuecomment-620042918

**Comment by dhmlau** [comment]
We just switch the contribution method from CLA to DCO, making your contribution easier in the future. Please sign the commits with DCO by amending your commit messages with `-s` flag and push the changes again. If you're using the command line, you can:

```
git commit --amend -s
git push --force-with-lease
```

Please refer to [this docs page](https://loopback.io/doc/en/contrib/code-contrib.html#developer-certificate-of-origin-dco) for details. Thanks!
